linode/vpc/framework_models.go

```diff
@@ -0,0 +1,177 @@
+package vpc
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper/customtypes"
+)
+
+type VPCResourceModel struct {
+	ID                   types.Int64                        `tfsdk:"id"`
+	Label                types.String                       `tfsdk:"label"`
+	Description          types.String                       `tfsdk:"description"`
+	Region               types.String                       `tfsdk:"region"`
+	Subnets              types.List                         `tfsdk:"subnets"`
+	Created              customtypes.RFC3339TimeStringValue `tfsdk:"created"`
+	Updated              customtypes.RFC3339TimeStringValue `tfsdk:"updated"`
+	SubnetsCreateOptions []VPCSubnetCreateOpts              `tfsdk:"subnets_create_options"`
+}
+
+type VPCDataSourceModel struct {
+	ID          types.Int64                        `tfsdk:"id"`
+	Label       types.String                       `tfsdk:"label"`
+	Description types.String                       `tfsdk:"description"`
+	Region      types.String                       `tfsdk:"region"`
+	Subnets     types.List                         `tfsdk:"subnets"`
+	Created     customtypes.RFC3339TimeStringValue `tfsdk:"created"`
+	Updated     customtypes.RFC3339TimeStringValue `tfsdk:"updated"`
+}
+
+type VPCSubnetCreateOpts struct {
+	Label types.String `tfsdk:"label"`
+	IPv4  types.String `tfsdk:"ipv4"`
+}
+
+func (d *VPCResourceModel) parseComputedAttributes(
+	ctx context.Context,
+	vpc *linodego.VPC,
+) diag.Diagnostics {
+	d.ID = types.Int64Value(int64(vpc.ID))
+	d.Description = types.StringValue(vpc.Description)
+
+	if vpc.Created != nil {
+		d.Created = customtypes.RFC3339TimeStringValue{
+			StringValue: types.StringValue(vpc.Created.Format(time.RFC3339)),
+		}
+	} else {
+		d.Created = customtypes.RFC3339TimeStringValue{
+			StringValue: types.StringNull(),
+		}
+	}
+
+	if vpc.Updated != nil {
+		d.Updated = customtypes.RFC3339TimeStringValue{
+			StringValue: types.StringValue(vpc.Updated.Format(time.RFC3339)),
+		}
+	} else {
+		d.Updated = customtypes.RFC3339TimeStringValue{
+			StringValue: types.StringNull(),
+		}
+	}
```

Comment:
Just a nitpick, but we have a helper function to simplify this logic a bit 🙂 

```suggestion
	d.Created = customtypes.RFC3339TimeStringValue{
		StringValue: helper.NullableTimeToFramework(vpc.Created),
	}

	d.Updated = customtypes.RFC3339TimeStringValue{
		StringValue: helper.NullableTimeToFramework(vpc.Updated),
	}
```